lib/Parse/ParseDecl.cpp

```diff
@@ -3055,6 +3055,14 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
       AttrRange = SourceRange(Loc);
     }
 
+    // Embedded Swift prohibits weak/unowned but allows unowned(unsafe).
```

Comment:
Instead of enforcing this in the parser, let’s do it in type resolution. That way ASTGen won’t need to duplicate this check.

Reply:
Does this look good now?